apps/meteor/client/views/account/security/TwoFactorTOTP.tsx

```diff
@@ -51,37 +51,47 @@ const TwoFactorTOTP = (props: TwoFactorTOTPProps): ReactElement => {
 		updateCodesRemaining();
 	}, [checkCodesRemainingFn, setCodesRemaining, totpEnabled]);
 
-	const handleEnableTotp = useCallback(async () => {
-		try {
-			const result = await enableTotpFn();
-
-			setTotpSecret(result.secret);
-			setQrCode(qrcode(result.url, { size: 200 }));
+	const handleEnableTotp = useCallback(
```

Comment:
Even thought you don't have separate buttons to enable/disable totp, it doesn't mean it all needs to be combined within the same function. You can still have two separate functions one for enabling and another for disabling and then call then accordingly. Eg.
```js
const handleToggleTotp = (e) => {
  if (e.currentTarget.checked) {
    enableTotp();
  } else {
    disableTotp();
  }
}
```
The way it is code readability has suffered imo.

Reply:
`useMutation` could also be an option for managing success and error states.